### PR TITLE
[16.01] Slight tweaks to our tabix usage. 

### PR DESCRIPTION
**Pull request by dannon**
This fixes trackster without more divergence in our version of pysam.

### CODE REVIEW

**Comment by dannon** [comment]
Toolshed test failed because the toolshed is temporarily down, unrelated to these changes.

**Comment by afgane** [comment]
Loading up a BED file works initially, but as I zoom in, nothing shows up in the display and the following shows up in the log:

```
162.129.251.86 - - [09/Mar/2016:16:23:40 +0000] "GET /api/genomes/hg38?data_type=data&chrom=chr22&low=19986336&high=20029410&mode=Auto&resolution=107.68596352101505&reference=true HTTP/1.0" 500 - "http://54.145.144.226/visualization/trackster?dataset_id=bbbfa414ae315caf&dbkey=hg38&hda_ldda=hda" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.109 Safari/537.36"
Error - <type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'sequence'
URL: http://54.145.144.226/api/genomes/hg38?data_type=data&chrom=chr22&low=19986336&high=20029410&mode=Auto&resolution=107.68596352101505&reference=true
File '/mnt/galaxy/galaxy-app/lib/galaxy/web/framework/middleware/error.py', line 151 in __call__
  app_iter = self.application(environ, sr_checker)
File '/mnt/galaxy/galaxy-app/.venv/local/lib/python2.7/site-packages/paste/recursive.py', line 85 in __call__
  return self.application(environ, start_response)
File '/mnt/galaxy/galaxy-app/.venv/local/lib/python2.7/site-packages/paste/httpexceptions.py', line 640 in __call__
  return self.application(environ, start_response)
File '/mnt/galaxy/galaxy-app/lib/galaxy/web/framework/base.py', line 126 in __call__
  return self.handle_request( environ, start_response )
File '/mnt/galaxy/galaxy-app/lib/galaxy/web/framework/base.py', line 184 in handle_request
  body = method( trans, **kwargs )
File '/mnt/galaxy/galaxy-app/lib/galaxy/web/framework/decorators.py', line 48 in call_and_format
  return dumps( func( self, trans, *args, **kwargs ), **json_kwargs )
File '/mnt/galaxy/galaxy-app/lib/galaxy/webapps/galaxy/api/genomes.py', line 42 in show
  rval = { 'dataset_type': 'refseq', 'data': region.sequence }
AttributeError: 'NoneType' object has no attribute 'sequence'
```

File: lib/galaxy/visualization/data_providers/genome.py [hunk 1]
```diff
@@ -347,13 +346,15 @@ class TabixDataProvider( FilterableMixin, GenomeDataProvider ):
 
     def open_data_file( self ):
         return ctabix.Tabixfile(self.dependencies['bgzip'].file_name,
-                                index_filename=self.converted_dataset.file_name)
+                                index=self.converted_dataset.file_name)
 
     def get_iterator( self, data_file, chrom, start, end, **kwargs ):
-        start, end = int(start), int(end)
+        # chrom must be a string (not unicode), start/end integers.
```
Comment (mvdbeek) [inline lib/galaxy/visualization/data_providers/genome.py]: can you add that this is due to pysam version xx requiring this? that way in the future we can check whether or not we can use unicode again.

Comment (dannon) [inline lib/galaxy/visualization/data_providers/genome.py]: Sure.

**Comment by dannon** [comment]
@afgane Hrmm.  This a cloud instance?  I'm not sure if that might be something different happening with the genome configuration there, I'm not able to reproduce that.

**Comment by afgane** [comment]
OK, let me dig around a bit.

**Comment by dannon** [comment]
(Followup on @afgane's error -- was an issue with the cloud instance he was testing on, unrelated to these changes)

**Comment by mvdbeek** [comment]
I tested it as well - indeed, it was broken before and fixed with the patch. :+1:

File: lib/galaxy/visualization/data_providers/genome.py [hunk 2]
```diff
@@ -347,13 +346,16 @@ class TabixDataProvider( FilterableMixin, GenomeDataProvider ):
...
     def open_data_file( self ):
         return ctabix.Tabixfile(self.dependencies['bgzip'].file_name,
-                                index_filename=self.converted_dataset.file_name)
+                                index=self.converted_dataset.file_name)
 
     def get_iterator( self, data_file, chrom, start, end, **kwargs ):
-        start, end = int(start), int(end)
+        # chrom must be a string, start/end integers.
+        # in previous versions of pysam, unicode was accepted for chrom, but not in 8.4
+        chrom = str(chrom)
```
Comment (nsoranzo) [inline lib/galaxy/visualization/data_providers/genome.py:354]: Will this work in Python 3?

Comment (dannon) [inline lib/galaxy/visualization/data_providers/genome.py:354]: I don't think it'll blow up, but you're more than welcome to test it ;)

**Comment by nturaga** [comment]
+1 This works

**Comment by davebx** [comment]
@galaxybot test this

**Pull request merged**